src/run.ts

```diff
@@ -286,9 +292,21 @@ export async function runVersion({
     let cmd = semverLt(changesetsCliPkgJson.version, "2.0.0")
       ? "bump"
       : "version";
-    await exec("node", [resolveFrom(cwd, "@changesets/cli/bin.js"), cmd], {
-      cwd,
-    });
+    await exec(
+      "node",
+      [
+        fileURLToPath(
+          resolve(
+            "@changesets/cli/bin.js",
+            pathToFileURL(path.join(cwd, "x.cjs")).toString()
+          )
+        ),
```

Comment:
this is... quite a dance to make it work. Not the biggest issue but I wish there would be a builtin way to make this easier. 

If I'm not mistaken, it also means that a bug was introduced here:
https://github.com/changesets/changesets/blob/268a29fedc948f22c672a3b1e3e51df4427f478d/packages/cli/src/commit/getCommitFunctions.ts#L16

A directory can't be used directly as input to `resolve`. It should be a filepath. cc @bluwy for double-checking my logic

Reply:
Yeah I think you're right. That should probably be a file path otherwise it resolves from one directory higher, which probably coincidentally worked because there's no `node_modules` in `.changeset/` for the linked code.

About this code, could we also use `require.resolve` instead of `import-meta-resolve` too?

Reply:
> About this code, could we also use require.resolve instead of import-meta-resolve too?

Yep, that was my plan - I just wanted to have a pushed out reference of how to do this using `import-meta-resolve` ;p I simplified this now.